packages/extras/src/lib/components/Environment/Environment.svelte

```diff
@@ -0,0 +1,133 @@
+		if (isCubeMap && format == 'ldr') return new CubeTextureLoader()
+		if (!isCubeMap && format == 'ldr') return new TextureLoader()
+		if (isCubeMap && format == 'hdr') return new HDRCubeTextureLoader()
+		if (!isCubeMap && format == 'hdr') return new RGBELoader()
+		return new TextureLoader()
+	}
+
+	const loadEnvironment = () => {
+		if (!renderer) throw new Error('Threlte renderer undefined')
+		const pmremGenerator = new PMREMGenerator(renderer)
+		DefaultLoadingManager.onLoad = function () {
+			pmremGenerator.dispose()
+		}
+		const loader: any = pickLoader()
+
+		if (isCubeMap) {
+			pmremGenerator.compileCubemapShader()
+			loader.setPath(path || '').load(files, (texture: any) => {
+				currentEnvMap = texture
+				texture.encoding = encoding || sRGBEncoding
+				currentProcessedEnv = pmremGenerator.fromCubemap(texture).texture
+				setEnvironment()
+			})
+		} else {
+			pmremGenerator.compileEquirectangularShader()
+			loader.load(`${path || ''}${files}`, (texture: any) => {
+				currentEnvMap = texture
+				texture.encoding = encoding || LinearEncoding
+				currentProcessedEnv = pmremGenerator.fromEquirectangular(texture).texture
+				setEnvironment()
+			})
+		}
+		currentFormat = format
+	}
+
+	const setEnvironment = () => {
+		scene.environment = currentProcessedEnv
+		if (isBackground) scene.background = currentProcessedEnv
+		invalidate()
+	}
+
+	const toggleGroundEnv = (
+		groundEnv: GroundProjectedEnv | undefined,
+		groundEnvProps: EnvironmentProperties['groundProjection'],
+		envMap: Texture
+	) => {
+		if (groundEnv && !groundEnvProps) {
+			scene.remove(currentGroundEnv)
+			currentGroundEnv = undefined
+			invalidate('Removing ground projected environment')
+		}
+		if (!groundEnv && groundEnvProps && envMap) {
+			currentGroundEnv = new GroundProjectedEnv(envMap)
+			currentGroundEnv.scale.setScalar(groundEnvProps.scalar)
+			currentGroundEnv.radius = groundEnvProps.radius
+			currentGroundEnv.height = groundEnvProps.height
+			scene.add(currentGroundEnv)
+			invalidate('Enabling ground projected environment')
+		}
+		if (groundEnv && groundEnvProps) {
+			currentGroundEnv.radius = groundEnvProps.radius
+			currentGroundEnv.height = groundEnvProps.height
+			invalidate('Updating ground projected environment properties')
+		}
+	}
+
+	$: toggleGroundEnv(currentGroundEnv, groundProjection, currentEnvMap)
+
+	$: {
+		// Reload environment on path change
+		if (envPath != previousEnvPath || format != currentFormat) {
+			if (currentProcessedEnv) {
+				currentProcessedEnv.dispose()
+			}
+			loadEnvironment()
+			previousEnvPath = envPath
+		}
+
+		if (!isBackground && scene.background) {
+			scene.background = null
+			invalidate('Removing Environment as scene.background')
+		}
+		if (isBackground && !scene.background && currentProcessedEnv) {
+			scene.background = currentProcessedEnv
+			invalidate('Adding Environment as scene.background')
+		}
+	}
+
+	onDestroy(() => {
+		scene.environment = null
```

Comment:
Maybe we should save the previous (i.e. before this component was mounted) `scene.environment` and `scene.background` and `onDestroy` set it back to these values.